File: assignment1/Linear Search.c
#include <stdio.h>

int n, element, i;
int arr[1000];

void linearSearch(void);

int main(void)
{


  printf("\nEnter the number of elements: ");
  scanf("%d", &n);

  printf("\nEnter the elements: ");
  for(i = 0; i < n; i++)
  {
      scanf("%d", &arr[i]);
  }

  printf("\nEnter the element which you want to search: ");
  scanf("%d", &element);

  linearSearch();

  return 0;

}

void linearSearch()
{
    for(i = 0; i < n; i++)
    {
        if(arr[i] == element)
        {
            printf("\n%d is found in %d location.", element, i);
            break;
        }
    }

    if(i == n)
        printf("%d is not found in this array.", element);
}

File: assignment1/LinearSearch.java
import java.util.Scanner;
import java.util.Random;

public class LinearSearch{
    public int element;
    public int ar1[] = new int[1000];
    public int ar2[] = new int[1000];
    
    public LinearSearch(){
        this.element = 0;
        
    }
    
    public void randomNumber(){
      Random rn = new Random();
      for(int i = 0; i < ar1.length; i++){
            ar1[i] = rn.nextInt();
            System.out.println(ar1[i]);
        }
    }
    
    public void enterElement(){
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the element which you want to search: ");
        element = sc.nextInt();
    }
   
    public void linearSearch()
{
    int flag, i, j;
    flag = -1;
    for(i = 0, j = 0; i < ar1.length; i++)
    {
        if(ar1[i] == element)
        {
            ar2[j] = i;
            flag = 1;
            j++;
        }
    }

    if(flag == -1)
        System.out.println(element + " is not found in this array." );
    else
    {
        System.out.println(element + " is found in the following location(s).");
        for(i = 0; i < j; i++){
            System.out.println(ar2[i] + " ");
        }
    }
}
   
}